### PR TITLE
Add questionnaire report

### PR DESCRIPTION
**Pull request by carmeltsadok**

### CODE REVIEW

File: mockserver/report/report.manager.js [hunk 1]
```diff
@@ -54,6 +54,61 @@ class ReportManager {
     "{""questionnaire"":[""60ae7ce0271ed648d4a38fef"",""60af5a0dc323bb047cdf137e""],""notes"":[{""_id"":""608ec9d5ccdb0a474493c7de"",""wroteBy"":""606ac716585a4237379a978d"",""note"":""koksinel met""},{""_id"":""608ec9e4ccdb0a474493c7e0"",""wroteBy"":""606ac716585a4237379a978d"",""note"":""koksinel met""}],""videoUrl"":""asdf.com""}","[""60242603973fa549bc148900"",""606ac716585a4237379a978d"",""1200007b973fa549bc148927""]","kirya","2021-05-31T06:47:41.818Z","קליינאוורדמן","נעם יהונתן","324828060","123123123","node-1"
     "{""questionnaire"":[""60ae7ce0271ed648d4a38fef"",""60af5a0dc323bb047cdf137e""],""notes"":[{""_id"":""608ec9d5ccdb0a474493c7de"",""wroteBy"":""606ac716585a4237379a978d"",""note"":""koksinel met""},{""_id"":""608ec9e4ccdb0a474493c7e0"",""wroteBy"":""606ac716585a4237379a978d"",""note"":""koksinel met""}],""videoUrl"":""asdf.com""}","[""60242603973fa549bc148900"",""606ac716585a4237379a978d"",""1200007b973fa549bc148927""]","kirya","2021-05-31T06:47:41.818Z","קליינאוורדמן","נעם יהונתן","324828060","123123123","node-1"`);
   }
+
+  static async createQuestionnaireReport(_req, res) {
```
Comment (PCOffline) [inline mockserver/report/report.manager.js:58]: Remove useless async
```suggestion
  static createQuestionnaireReport(_req, res) {
```

File: src/pages/Reports/components/EventsInputs/EventsInputs.jsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+ }, ref) => {
+  const classes = useStyles();
+  const { t } = useTranslation();
+  const currentUser = UserStore.userProfile;
+  const [selectedNodeGroups, setSelectedNodeGroups] = useState([]);
+  const [selectedUnits, setSelectedUnits] = useState([]);
+
+  useEffect(() => {
+    setCanSubmit(
+      !(nodeGroupsRequiredRoles.includes(currentUser.role) && !selectedNodeGroups.length),
```
Comment (PCOffline) [inline src/pages/Reports/components/EventsInputs/EventsInputs.jsx:24]: Change to a more intuitive syntax
```suggestion
      !nodeGroupsRequiredRoles.includes(currentUser.role) || selectedNodeGroups.length,
```

File: src/pages/Reports/components/QuestionnaireInputs/QuestionnaireInputs.jsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  const currentUser = UserStore.userProfile;
+  const [selectedNode, setSelectedNode] = useState();
+  const [selectedTarget, setSelectedTarget] = useState();
+  const [selectedNodeGroup, setSelectedNodeGroup] = useState();
+  const [selectedUnit, setSelectedUnit] = useState();
+  const [nodes, setNodes] = useState([]);
+
+  useEffect(() => {
+    setCanSubmit(
+      !(nodeGroupsRequiredRoles.includes(currentUser.role) && !selectedNodeGroup)
```
Comment (PCOffline) [inline src/pages/Reports/components/QuestionnaireInputs/QuestionnaireInputs.jsx:31]: Change to a more intuitive syntax
```suggestion
      !nodeGroupsRequiredRoles.includes(currentUser.role) || selectedNodeGroup
```

**Review by PCOffline — CHANGES_REQUESTED** [review]

File: src/pages/Reports/components/EventsInputs/EventsInputs.jsx (hunk 2 above)
Comment (carmeltsadok) [reply src/pages/Reports/components/EventsInputs/EventsInputs.jsx:24]: it is more readable that way

File: src/pages/Reports/components/QuestionnaireInputs/QuestionnaireInputs.jsx (hunk 3 above)
Comment (carmeltsadok) [reply src/pages/Reports/components/QuestionnaireInputs/QuestionnaireInputs.jsx:31]: it is more readable that way

**Review by PCOffline — APPROVED** [review]

File: src/pages/Reports/components/QuestionnaireInputs/QuestionnaireInputs.jsx (hunk 3 above)
Comment (PCOffline) [reply src/pages/Reports/components/QuestionnaireInputs/QuestionnaireInputs.jsx:31]: How?

**Review by maykremer — APPROVED** [review]

**Pull request merged**